### PR TITLE
fix(Scripts/ZulGurub): Improvements to Gah'zranka:

### LINKED ISSUE #11560: [ZG] Gahz'ranka - Mudskunk Lure / Issues with summoning
### Current Behaviour

- The lure is thrown the at the wrong place.
- The water splash visual effect happens immediately, and the gameobject Mudskunk Lure does not appear.
- No roaring sound and screen shake visual effect.
- Gahz'ranka spawns way too fast and way too far inland.
- Gahz'ranka has no movement once it spawns.
- Gahz'ranka can be summoned several times.

https://user-images.githubusercontent.com/100873791/165771296-3f8cd7db-b234-49e8-b95d-2ed5c56cd12a.mp4

note: I'm using the wrong Game Client Data files in this video, that's why the colors are different.

### Expected Blizzlike Behaviour

The lure is thrown near where the boss should spawn, making a Mudskunk Lure object appear with a water splash,  and after a couple of seconds (between 5 and 10) another water splash effect occurs, the object despawns, Gahz'ranka spawns, the roar is heard and the screen shakes.
Gahz'ranka spawns at the bottom of the lake, moving towards the middle and onto Pagle's Pointe.
Gahz'ranka should only be spawned once per reset, if he despawns it should not be possible to summon him again. He despawns after a wipe.

### Source

https://youtu.be/JEmFTdr65UY?t=33 Probably best shows the path Gahz takes
https://www.youtube.com/watch?v=1u2-TiUISDY Probably best shows where the boss should stop
https://youtu.be/LeIGH7ZpUts?t=100 also shows where 'ranka should stop
https://youtu.be/BNsf4SbHFo4?t=267
https://youtu.be/T4NaB7KU0jQ?t=4

>Multiple summons:  No, you can only summon him once per saved instance.  My raid summoned him and we had the entire core 8 people DCed at once and caused a wipe(all raiding from one guys house and the cable modem burped).  When we got back and rezed we tried again and it failed.  After a petition to a GM I was told that he can only be summoned once per instance. ([source](https://classic.wowhead.com/npc=15114/gahzranka#comments:id=2801924))

### Steps to reproduce the problem

1. `.additem 19975 5`
2. `.additem 19974`
3. `.go xyz -11695 -1770 13 309` Teleports to Pagle's Pointe
4. Use Mudskunk Lure and observe Gahz'ranka spawning

### Extra Notes

The Mudskunk Lure gameobject should spawn somewhere near `X: -11683.5 Y: -1723.5 Z: 9` but this isn't exact and it's just from my observation by eye, not real data.
Mudskunk Lure Object ID: 180346

### AC rev. hash/commit

https://github.com/azerothcore/azerothcore-wotlk/commit/59e45c251eb53abb0ed4b37a57254cbdf415e69a

### Operating system

Windows 10

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by UltraNix**
Summoned Mudskunk Lure object and added visual splash.
Added pathing to Gahzranka
Corrected visual coordinates.
Gahzranka can be summoned only once per instance.
Fixes #11560

## Issues Addressed:

- Closes #11560

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

`.additem 19975 5`
`.additem 19974`
`.go xyz -11695 -1770 13 309` Teleports to Pagle's Pointe
Use Mudskunk Lure and observe Gahz'ranka spawning

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
- [ ] The thrown item and the splash are not at the same place as the Lure. I also think the Lure object spawns a tad too fast and would appear before the splash

 - [ ] There's weird pathing for Gahz'ranka, which would make it easily exploitable, M'Dic should know how to fix it, but I think it has to do with CTM and UnitFlags

Moreover, I think the pathing issue in water is a regression, because I remember *not* seeing that when I tested the boss in April 28th

https://user-images.githubusercontent.com/100873791/170882512-2734ebe8-1f74-4c7e-8b51-d154a2abfda8.mp4

**Comment by UltraNix** [comment]
@Gultask Shoud be okay now.

**Comment by Maelthyrr** [comment]
> @Gultask Shoud be okay now.

For the  thrown item, it's ok, but the pathway stay ugly. He goes often through the map. 

I've found another video if you want where he go directly to the land and doesn't turn around in the water.

https://youtu.be/LeIGH7ZpUts?t=95

**Comment by UltraNix** [comment]
Okay - will revert the changes to pathinding.
It's a general problem there with mmaps, not with this mob.

**Comment by Nefertumm** [comment]
conflicts and you should add the include for spellscript

File: data/sql/updates/pending_db_world/rev_1653819851783034500.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--
+UPDATE `spell_target_position` SET `PositionX`=-11688.5, `PositionY`=-1737.74, `PositionZ`=8.409842 WHERE id IN (24325, 24593);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1653819851783034500.sql:2]:
```suggestion
UPDATE `spell_target_position` SET `PositionX`=-11688.5, `PositionY`=-1737.74, `PositionZ`=8.409842 WHERE `id` IN (24325, 24593);
```

File: data/sql/updates/pending_db_world/rev_1653819851783034500.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+DELETE FROM `event_scripts` WHERE `id`=9104;
+
+DELETE FROM `spell_script_names` WHERE `spell_id`=24325;
+INSERT INTO `spell_script_names` VALUES
+(24325, "spell_pagles_point_cast");
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1653819851783034500.sql:7]:
```suggestion
(24325, 'spell_pagles_point_cast');
```

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_gahzranka.cpp [hunk 3]
```diff
@@ -32,8 +32,8 @@ enum Spells
     SPELL_FROSTBREATH               = 16099,
     SPELL_MASSIVEGEYSER             = 22421,
     SPELL_SLAM                      = 24326,
-    SPELL_THRASH                    = 3417 // Triggers 3391
-};
+    SPELL_THRASH                    = 3417, // Triggers 3391
+    SPELL_SPLASH                    = 24593
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_gahzranka.cpp:36]:
```suggestion
    SPELL_SPLASH                    = 24593
};
```

**Review by Nefertumm — APPROVED** [review]

**Pull request merged**